Acceptance test for RecorderAlertHandler

neomatrix369 opened the pull request:
Added the VideoRecorderAlertHandler using the VideoRecorderStartedEvent, implemented the acceptance test with couple of video files and a video url check

@julianghionoiu We need to discuss API for the Video event. Wasn't clear from the sample event provided, how much of it would be available when writing this piece of work.

We need to discuss how we will identify from the S3 Put event if this put action is for the Video started event - do you do the wiring on the AWS end?

neomatrix369 committed:
Added the VideoRecorderAlertHandler using the VideoRecorderStartedEvent, implemented the acceptance test with couple of video files and a video url check

neomatrix369 commented on build.gradle:
```diff
@@ -29,15 +30,25 @@ dependencies {
 
-            'ro.ghionoiu:queue-client:0.1.17',
-            
+//            'ro.ghionoiu:queue-client:0.1.17',
```
This will need to go when all events are implemented and finalised

neomatrix369 commented on src/main/java/tdl/datapoint/infra_events/VideoRecorderAlertHandler.java:
```diff
@@ -84,12 +84,16 @@ public String handleRequest(Map<String, Object> s3EventMap, Context context) {
         String challengeId = event.getChallengeId();
+        String participantId = event.getParticipantId();
+
+        participantEventQueue.send(new VideoRecorderStartedEvent(System.currentTimeMillis(),
```
We need to discuss here what the API should be

julianghionoiu commented on src/main/java/tdl/datapoint/infra_events/VideoRecorderAlertHandler.java:
```diff
@@ -23,8 +23,8 @@
 
-public class SourceCodeUploadHandler implements RequestHandler<Map<String, Object>, String> {
-    private static final Logger LOG = Logger.getLogger(SourceCodeUploadHandler.class.getName());
+public class VideoRecorderAlertHandler implements RequestHandler<Map<String, Object>, String> {
```
This will be one single generic handler, the body of the method is responsible for detecting the event type.

neomatrix369 replied:
Ok got it, I was thinking of getting one of them to work first before trying to make for something for all of them - I'll have to see how this goes

julianghionoiu commented on src/test/java/tdl/datapoint/infra_events/support/TestVideoFile.java:
```diff
@@ -0,0 +1,17 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class TestVideoFile {
```
No need for this. Just a crafted JSON event.

neomatrix369 replied:
Ok

julianghionoiu commented on src/test/java/tdl/datapoint/infra_events/support/TestVideoFile.java:
```diff
@@ -0,0 +1,17 @@
+    private final Path resourcePath;
+
+    public TestVideoFile(String name) {
+        resourcePath = Paths.get("src/test/resources/", name);
```
Also, no need for the videos.

neomatrix369 replied:
Ok

neomatrix369 committed:
Applied code review comments and refactored code towards expected direction

julianghionoiu commented on src/test/java/tdl/datapoint/infra_events/RecordingsDatapointAcceptanceTest.java:
```diff
@@ -103,37 +100,40 @@ public void tearDown() throws Exception {
                 endsWith(participantId)));
 
         // When - Another upload event happens
-        S3Event s3Event2 = localS3Bucket.putObject(video2.asFile(), s3destination);
```
No need for a second event, it is not proving anything.

neomatrix369 replied:
Ok will remove redundant check

julianghionoiu commented on src/main/java/tdl/datapoint/infra_events/EventsAlertHandler.java:
```diff
@@ -73,9 +73,9 @@ private static AmazonSQS createSQSClient(String serviceEndpoint, String signingR
+    public String handleRequest(Map<String, Object> eventMap, Context context) {
         try {
-            handleS3Event(S3BucketEvent.from(s3EventMap, jsonObjectMapper));
+            handleS3Event(S3BucketEvent.from(eventMap, jsonObjectMapper));
```
Some event detection needs to happen at this point, some tests need to be added for this

neomatrix369 replied:
This part is still up in the air so will give it a think

neomatrix369 replied:
Mostly done via d1ca431

julianghionoiu reviewed (commented):
It is going in the right direction

neomatrix369 commented:
Do we want to wrap the event JSON inside the RecorderStartedEvent, or just extract the essential info i.e. `challengeId` and `participantId`

neomatrix369 committed:
- Removed S3 client implementation
- Detect if the event is an S3 event that has come in, if so, handle it
- Added logic to wrap S3 event into an SNS event
- Removed redundant logic as per code review comments

neomatrix369 committed:
Added a test for unsupported events - no event is sent to the SQS queue, updated some comments in the AWSEventsDatapointAcceptanceTest test and renamed the test class to reflect the testing being performed

julianghionoiu commented on src/test/java/tdl/datapoint/infra_events/AWSEventsDatapointAcceptanceTest.java:
```diff
@@ -103,14 +104,35 @@ public void start_recorder() throws Exception {
     }
 
+    @Test
+    public void start_something_unsupported_no_event_should_flow_through_the_sqs_server() throws Exception {
```
I think there are two behaviours here:
- unrecognised events should not flow to SQS
- unrecognised events should raise exception

julianghionoiu reviewed (commented):
Good progress

neomatrix369 committed:
Raise an exception if the system tries to send an unidentified event through to the SQS queue

neomatrix369 committed:
Removing references on S3 from the README and the .gitmodules file, also from comments in the code

neomatrix369 committed:
Now the lambda can be executed from the CLI via the serverless command via a provided JSON in the test resource (manual step in addition the passing test)

neomatrix369 committed:
Update queue-client's version to build the right events
